pkg/sql/schema_changer_test.go

```diff
@@ -2372,6 +2425,17 @@ CREATE TABLE t.test (k INT PRIMARY KEY, v INT);
 	if err := checkTableKeyCount(ctx, kvDB, 3, maxValue); err != nil {
 		t.Fatal(err)
 	}
+
+	// The notify schema change channel must be nil-ed out, or else
+	// running scrub will cause it to trigger again on an already closed
+	// channel.
+	// FIXME(joey): It seems odd that we need to do this. As far as I
+	// know, the knob is getting run because a new transaction is make
+	// when running the following statement. Does this seem correct?
+	notifySchemaChange = nil
```

Comment:
This is because the filter gets called even if the sql statement is not involved with a schema change. I'm not opposed to leave it the way you have.